src/libaktualizr/uptane/secondaryinterface.h

```diff
@@ -20,21 +20,32 @@ namespace Uptane {
 
 class SecondaryInterface {
  public:
+  // This ctor should be removed as the secondary configuration SecondaryConfig
+  // is the secondaries's specific, see SecondaryConfig declaration
   explicit SecondaryInterface(SecondaryConfig sconfig_in) : sconfig(std::move(sconfig_in)) {}
   virtual ~SecondaryInterface() = default;
+  // not clear what this method for, can be removed
   virtual void Initialize(){};  // optional step, called after device registration
+  // should be pure virtual, since the current implementation reads from the secondaries specific config
   virtual EcuSerial getSerial() { return Uptane::EcuSerial(sconfig.ecu_serial); }
+  // should be pure virtual, since the current implementation reads from the secondaries specific config
   virtual Uptane::HardwareIdentifier getHwId() { return Uptane::HardwareIdentifier(sconfig.ecu_hardware_id); }
   virtual PublicKey getPublicKey() = 0;
 
+  // getSerial(), getHwId() and getPublicKey() can be moved to seperate interface
+  // since their usage pattern differ from the following methods' one
```

Comment:
Separate interface? What do you mean?

Reply:
Well, serial, hw_id and pub_key are constant properties of a secondary and can be determined/obtained before or during a secondary object instantiation, while all other methods represent RPC calls that actually transfer messages between Primary and Secondary. I don't think, that there is  any practical need/benefit in breaking the current interface into two parts because of the aforementioned, but it might appear in the future.